### PR TITLE
fix(datastore) Set orchestrator status when no API configured

### PR DESCRIPTION
**Pull request by rjuliano**
If there is no API configured (i.e. DataStore is operating in local-only mode), calls to the DataStore end up timing out. This PR addresses this issue and modified tests to account for the scenario.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java [hunk 1]
```diff
@@ -102,9 +102,20 @@ public void configureAndInitializeInLocalMode() throws AmplifyException {
         //Configure DataStore with an empty config (All defaults)
         ApiCategory emptyApiCategory = spy(ApiCategory.class);
         AWSDataStorePlugin standAloneDataStorePlugin = new AWSDataStorePlugin(modelProvider, emptyApiCategory);
+        SynchronousDataStore synchronousDataStore = SynchronousDataStore.delegatingTo(standAloneDataStorePlugin);
         standAloneDataStorePlugin.configure(new JSONObject(), context);
         standAloneDataStorePlugin.initialize(context);
+
+        // Trick the DataStore since it's not getting initialized as part of the Amplify.initialize call chain
+        Amplify.Hub.publish(HubChannel.DATASTORE, HubEvent.create(InitializationStatus.SUCCEEDED));
+
```
Comment (raphkim) [inline aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java:111]: why is this necessary on top of the actual call to initialization?

Comment (rjuliano) [reply aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java:111]: Because the message only gets sent from the Amplify Category class I believe. For unit testing purposes, I'm initializing the plugin directly

Comment (raphkim) [reply aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java:111]: Ah, makes sense. 👍
